examples/general/abs_home_path/src/lib.rs

```diff
@@ -162,3 +177,19 @@ fn ui() {
         .rustc_flags(["--test"])
         .run();
 }
+
+// Test to verify that lint isn't triggered in build scripts
+#[test]
+#[allow(clippy::assertions_on_constants)]
+fn build_script_detection_works() {
```

Comment:
We will need a test eventually, but I don't think this tests what we want.

The test we want would have a build script with `env!("CARGO_MANIFEST_PATH")` or something like that.

Actually, here's an example in Dylint: https://github.com/trailofbits/dylint/blob/5aad720e3cb54c93dfa2ad7c0d5e7a0cb29c3cf4/dylint/build.rs#L23-L24

I put you on the spot to add a test, which was unfair.

Could I suggest removing this test and we just merge with no test?